Content.Client/UserInterface/GhostGui.cs

```diff
@@ -18,18 +25,133 @@ public GhostGui(GhostComponent owner)
+                }
+            }));
 
             Update();
         }
 
+
         public void Update()
         {
             ReturnToBody.Disabled = !_owner.CanReturnToBody;
         }
     }
+
+    internal class GhostTargetWindow : SS14Window
+    {
+
+        protected override Vector2? CustomSize => (300, 450);
+        private readonly IPlayerManager _playerManager;
+        private List<Button> _buttons = new List<Button>();
+        private readonly GhostComponent _owner;
+        private VBoxContainer _buttonContainer;
+
+        public GhostTargetWindow(IPlayerManager manager,GhostComponent owner)
+        {
+            Title = "Ghost Warp";
+            _playerManager = manager;
+            _owner = owner;
+            _owner.GhostRequestWarpPoint();
+            _owner.GhostRequestPlayerNames();
+
+            var margin = new MarginContainer()
+            {
+                SizeFlagsVertical = SizeFlags.FillExpand,
+                SizeFlagsHorizontal = SizeFlags.FillExpand,
+            };
+
+            _buttonContainer = new VBoxContainer()
+            {
+                SizeFlagsVertical = SizeFlags.FillExpand,
+                SizeFlagsHorizontal = SizeFlags.Fill,
+                SeparationOverride = 5,
+
+            };
+
+            var _scrollBarcontainer = new ScrollContainer()
+            {
+                Name = "_v_scroll",
+                SizeFlagsVertical = SizeFlags.FillExpand,
+                SizeFlagsHorizontal = SizeFlags.FillExpand
+            };
+
+            margin.AddChild(_scrollBarcontainer);
+            _scrollBarcontainer.AddChild(_buttonContainer);
+
+            Contents.AddChild(margin);
+
+        }
+
+        public void Populate()
+        {
+            AddButtonPlayer();
+            AddButtonLocations();
+            OpenCentered();
+        }
+
+        private void AddButtonPlayer()
+        {
+
+            foreach (var (key, value) in _owner.PlayerName)
+            {
+                _buttons.Add(new Button
+                {
+                    Text = value,
+                    TextAlign = Label.AlignMode.Right,
+                    SizeFlagsHorizontal = SizeFlags.ShrinkCenter,
+                    SizeFlagsVertical = SizeFlags.ShrinkCenter,
+                    SizeFlagsStretchRatio = 1,
+                    CustomMinimumSize = (230, 20),
+                    ClipText = true,
+                });
+                Button currentButtonRef = _buttons[^1];
+                currentButtonRef.OnPressed += (args) =>
+                {
+                    _owner.SendGhostWarpRequestMessage(key);
+                };
+                _buttonContainer.AddChild(currentButtonRef);
+            }
+        }
+        private void AddButtonLocations()
+        {
+            for (int i = 0; i < _owner.WarpName.Count-1; i++)
+            {
+                string name = _owner.WarpName[i];
+
+                _buttons.Add(new Button
+                {
+                    Text = name,
```

Comment:
Maybe add a Info here that this is a warp point and not a player? Like `Text = $"Warp: {name}"` or something like that